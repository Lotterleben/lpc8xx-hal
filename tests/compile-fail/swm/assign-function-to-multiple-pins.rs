use lpc8xx_hal::{
    Peripherals,
    pins::{
        self,
        Pin,
    },
    swm,
};


fn main() {
    let mut p = Peripherals::take().unwrap();

    let     swm    = p.SWM.split();
    let mut syscon = p.SYSCON.split();

    #[cfg(feature = "82x")]
    let mut swm_handle = swm.handle;
    #[cfg(feature = "845")]
    let mut swm_handle = swm.handle.enable(&mut syscon.handle);

    let pio0_0: Pin<_, pins::state::Unused> = p.pins.pio0_0;
    let pio0_1: Pin<_, pins::state::Unused> = p.pins.pio0_1;

    let u0_rxd: swm::Function<_, swm::state::Unassigned> =
        swm.movable_functions.u0_rxd;

    let (u0_rxd, _) = u0_rxd.assign(pio0_0.into_swm_pin(), &mut swm_handle);
    // Should fail: Already assigned function to another pin.
    let (u0_rxd, _) = u0_rxd.assign(pio0_1.into_swm_pin(), &mut swm_handle);
}
