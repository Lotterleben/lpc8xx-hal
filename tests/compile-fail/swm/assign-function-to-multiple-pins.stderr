error[E0599]: no method named `assign` found for struct `lpc8xx_hal::swm::Function<lpc8xx_hal::swm::U0_RXD, lpc8xx_hal::swm::state::Assigned<lpc8xx_hal::pins::PIO0_0>>` in the current scope
  --> $DIR/assign-function-to-multiple-pins.rs:30:30
   |
30 |     let (u0_rxd, _) = u0_rxd.assign(pio0_1.into_swm_pin(), &mut swm_handle);
   |                              ^^^^^^ help: there is an associated function with a similar name: `unassign`
